spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceProperties.java

```diff
@@ -118,11 +120,17 @@ public void setDynamicDestinations(String[] dynamicDestinations) {
 	public void setApplicationContext(ApplicationContext applicationContext)
 			throws BeansException {
 		this.applicationContext = (ConfigurableApplicationContext) applicationContext;
-		// override the bindings store with the environment-initializing version if in a
-		// Spring context
-		this.bindings = new EnvironmentEntryInitializingTreeMap<>(this.applicationContext,
-				BindingProperties.class, "spring.cloud.stream.default",
-				new TreeMap<String, BindingProperties>(String.CASE_INSENSITIVE_ORDER));
+	}
+
+	@Override
+	public void setEnvironment(Environment environment) {
```

Comment:
I am trying to understand how does this different from application context aware invocation (previous implemenation) as both `setEnvironment` and `setApplicationContext` are invoked by `ApplicationContextAwareProcessor` bean post processor?